docs/DeveloperGuide.md

```diff
@@ -8,23 +8,119 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## Implementation
+
+### Record and RecordList  
+
+The main class in our program is the ```Record``` and ```RecordList``` abstract classes, in which ```Income```, ```Expense``` will inherit from ```Record``` and ```IncomeList``` and ```ExpenseList``` will inherit from ```RecordList```. Most commands will act on instances of the ```Income```, ```Expense```,```IncomeList``` and ```ExpenseList``` classes.  
+
+![Record Class](images/Record_RecordList_UML_class.png)
+
```

Comment:
For this UML diagram:

1. Abstract class for RecordList and Record is incorrect. Instead of << >> , it should be denoted as {RecordList} instead?
2. The font size of the diagram is a little bit small, making it blurry and hard to read. Perhaps this could be changed to make it easier to read.